### ISSUE TITLE
ST3D on different classes

### ISSUE THREAD

**Issue by KevinDuBl** [CVMI-Lab/ST3D#49]
I did the experiments on Waymo -> MyOwnDataset of ['Vehicle', 'Pedestrian', 'Cyclist'] on PVRCNN. But the result shows, after self-train, the AP of the model for 3 classes is worse than that of the model just train for single. How can I get a stable model for all 3 classes?

**Comment by jihanyang** [comment]
It is hard to provide a general solution for all datasets. However, as I know, some researchers achieve amazing results on some custom datasets with multiple categories (approach fully supervised results) with a few adjustments to ST3D. Hence I believe that ST3D should work for other datasets with multiple categories, but need some tuning of hyper-parameters.

**Comment by KevinDuBl** [comment]
I fixed the bugs on multi-category and got good performance on multi-category. Thank you very much! Could you please tell when will you release the code for ST3D++?

**Comment by jihanyang** [comment]
Happy to heard that! We don't have a clear plan to release the code since the paper is still in submission. BTW, the techinical modification of ST3D++ is not significant, may be I can support DSNorm and SASD in this repo later. I believe that it should be easy to reproduce our results.

**Comment by KevinDuBl** [comment]
Thank you very much! I would like to see DSNorm and SASD as soon as possible!

**Comment by KevinDuBl** [comment]
For further research, I want to learn from the DSNorm and SASD. Could you please send me the code contains the DSNorm and SASD by email? Thank you very much!

**Comment by andrewcaunes** [comment]
Hello !
I am very interested in your solution for multi-category ST3D as I am working on a similar problem, is there any way you could send it to me by email please ?
Thanks a lot !

**Comment by KevinDuBl** [comment]
> Hello ! I am very interested in your solution for multi-category ST3D as I am working on a similar problem, is there any way you could send it to me by email please ? Thanks a lot !

You need to change the pred_labels for ignore in https://github.com/CVMI-Lab/ST3D/blob/68a9d3eb0a55c5adbb3d656c6fb597e29ad3155d/pcdet/utils/self_training_utils.py#L191 and others. And change the code for loading pseudo labels of multi-categories in https://github.com/CVMI-Lab/ST3D/blob/68a9d3eb0a55c5adbb3d656c6fb597e29ad3155d/pcdet/datasets/dataset.py#L152 and others.

**Comment by RhythmOfTheRain-Byte** [comment]
Excuse me, may i ask  ask how to modify the code specifically, applied to multi-class pre-training and multi-class ST3D training

**Comment by RhythmOfTheRain-Byte** [comment]
> > Hello ! I am very interested in your solution for multi-category ST3D as I am working on a similar problem, is there any way you could send it to me by email please ? Thanks a lot !
> 
> You need to change the pred_labels for ignore in
> 
> https://github.com/CVMI-Lab/ST3D/blob/68a9d3eb0a55c5adbb3d656c6fb597e29ad3155d/pcdet/utils/self_training_utils.py#L191
> 
> and others. And change the code for loading pseudo labels of multi-categories in
> https://github.com/CVMI-Lab/ST3D/blob/68a9d3eb0a55c5adbb3d656c6fb597e29ad3155d/pcdet/datasets/dataset.py#L152
> 
> and others.

Excuse me, may i ask ask how to modify the code specifically, applied to multi-class pre-training and multi-class ST3D training. Or just need to change the top two lines  pred_labels[ignore_mask] = -1   gt_names = np.array([self.class_names[0] for n in gt_boxes])

**Comment by RhythmOfTheRain-Byte** [comment]
> > Hello ! I am very interested in your solution for multi-category ST3D as I am working on a similar problem, is there any way you could send it to me by email please ? Thanks a lot !
> 
> You need to change the pred_labels for ignore in
> 
> https://github.com/CVMI-Lab/ST3D/blob/68a9d3eb0a55c5adbb3d656c6fb597e29ad3155d/pcdet/utils/self_training_utils.py#L191
> 
> and others. And change the code for loading pseudo labels of multi-categories in
> https://github.com/CVMI-Lab/ST3D/blob/68a9d3eb0a55c5adbb3d656c6fb597e29ad3155d/pcdet/datasets/dataset.py#L152
> 
> and others.

Excuse me, may i ask how to modify the code specifically, applied to multi-class pre-training and multi-class ST3D training. Or just need to change the top two lines pred_labels[ignore_mask] = -1 gt_names = np.array([self.class_names[0] for n in gt_boxes])
